bim2sim/elements/bps_elements.py

```diff
@@ -668,74 +668,38 @@ def bound_area(self) -> ureg.Quantity:
+        - VERTICAL: when normal is perpendicular to vertical (dot product ≈ 0)
+
+        Returns:
+            BoundaryOrientation: Enumerated orientation classification
         """
-        top_bottom = None
-        vertical = gp_XYZ(0.0, 0.0, 1.0)
-        # only assign top and bottom for elements, whose
-        # surface normals are not perpendicular to a vertical
-        if -1e-3 < self.bound_normal.Dot(vertical) < 1e-3:
-            top_bottom = "VERTICAL"
-        # is related bounds z-coordinate below current bound(self) then
-        # current bound is a floor (BOTTOM), otherwise a ceiling (TOP)
-        elif self.related_bound != None:
-            if (self.bound_center.Z() - self.related_bound.bound_center.Z()) \
-                    > 1e-2:
-                top_bottom = "BOTTOM"
-            elif (self.bound_center.Z() - self.related_bound.bound_center.Z()) \
-                    < -1e-2:
-                top_bottom = "TOP"
-            else:
-                # caution, this relies on correct surface normals
-                if vertical.Dot(self.bound_normal) < -0.8:
-                    top_bottom = "BOTTOM"
-                elif vertical.Dot(self.bound_normal) > 0.8:
-                    top_bottom = "TOP"
-        # for adiabatic bounds we need no tolerance
-        elif self.related_adb_bound is not None:
-            if self.bound_center.Z() > self.related_adb_bound.bound_center.Z():
-                top_bottom = "BOTTOM"
-            else:
-                top_bottom = "TOP"
-        # if no relating bound exists (exterior boundaries), we use the space
-        # center instead.
-        # TODO: This might fail for multi storey spaces.
-        else:
-            if (self.bound_center.Z() - self.bound_thermal_zone.space_center.Z()) \
-                    > 1e-2:
-                top_bottom = "TOP"
-            elif (self.bound_center.Z() - self.bound_thermal_zone.space_center.Z()) \
-                    < -1e-2:
-                top_bottom = "BOTTOM"
-            else:
-                # caution, this relies on correct surface normals
-                if vertical.Dot(self.bound_normal) < -0.8:
-                    top_bottom = "BOTTOM"
-                elif vertical.Dot(self.bound_normal) > 0.8:
-                    top_bottom = "TOP"
-        return top_bottom
-
-    # @staticmethod
-    # def compare_direction_of_normals(normal1, normal2):
-    #     """
-    #     Compare the direction of two surface normals (vectors).
-    #     True, if direction is same or reversed
-    #     :param normal1: first normal (gp_Pnt)
-    #     :param normal2: second normal (gp_Pnt)
-    #     :return: True/False
-    #     """
-    #     dotp = normal1.Dot(normal2)
-    #     check = False
-    #     if 1-1e-2 < dotp ** 2 < 1+1e-2:
-    #         check = True
-    #     return check
+        vertical_vector = gp_XYZ(0.0, 0.0, 1.0)
+        vertical_tolerance = 1e-3
+        cos_angle_top = math.cos(math.radians(89))
```

Comment:
this is covered by the default return of `return BoundaryOrientation.vertical`. But in fact, the check for `vertical_tolerance` is obsolete, as the default return statement will handle this anway. I adjusted the function, so that:

Greater than cos(89°) ≈ 0.017 → returns top
Less than cos(91°) ≈ -0.017 → returns bottom
If neither condition is met (meaning the dot product is between -0.017 and 0.017), it returns vertical

Changes done in 01b731702c863059f94ce002916f5a60a2f0be37